src/spikeinterface/core/template_tools.py

```diff
@@ -248,3 +249,165 @@ def get_template_extremum_amplitude(
         unit_amplitudes[unit_id] = extremum_amplitudes[unit_id][best_channel]
 
     return unit_amplitudes
+
+
+def move_templates(templates, displacements, source_probe, dest_probe=None, interpolation_method="cubic"):
+    """
+    Move all templates given some displacements using spatial interpolation (cubic or linear).
+    Optionally can be remapped to another probe with a diffrents geometry.
+
+    Before interpolation this function make is checking that templates can be interpolated on new
+    positions using the sparsity information : the entire sparse template must be covered at all new positions.
+    If the Templates object is dense then no check is possible and a wrning is emited.
+
+    Parameters
+    ----------
+    templates: Templates
+        A numpy array with dense templates.
+        shape = (num_template, num_sample, num_channel)
+    displacements: np.array
+        Displacement vector
+        shape: (num_displacement, 2)
+    source_probe: Probe
+        The Probe object on which templates are defined
+    dest_probe: Probe or None
+        The destination Probe. Can be different geometry than the original.
+        If None then the same probe  is used.
+    interpolation_method: str, default "cubic"
+        The interpolation method.
+    interpolation_method: str, default "cubic"
+        The interpolation method.
+    
+    Returns
+    -------
+    new_templates: list of Templates
+
+    """
+    if templates.are_templates_sparse():
+        # TODO make some check!!
+        pass
+    else:
+        warnings.warn("move_templates() with dense templates cannot checks that template can be interpolated at all places")
```

Comment:
```suggestion
        warnings.warn("move_templates() with dense templates cannot check that templates can be interpolated at all places")
```

I think the better preposition may be `to all places`, but you could interpolate at all places of the vector. So I think `at` if you want to emphasize the interpolation acting on the data vs `to` which emphasizes the result of the interpolation.